### PR TITLE
Add feature flag to allow enabling FileSystemAccess API.

### PR DESCRIPTION
**Pull request by mkarolin**
Fixes brave/brave-browser#18979

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by pes10k — APPROVED** [review]
lgtm

File: chromium_src/third_party/blink/public/common/features.h [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+namespace blink {
+namespace features {
+
+BLINK_COMMON_EXPORT extern const base::Feature kFileSystemAccessAPI;
+
+}  // namespace features
+}  // namespace blink
+
+
```
Comment (goodov) [inline chromium_src/third_party/blink/public/common/features.h:19]: unnecessary empty line

File: renderer/test/file_system_access_browsertest.cc [hunk 2]
```diff
@@ -49,14 +62,30 @@ class FileSystemAccessBrowserTest : public InProcessBrowserTest {
...
   ASSERT_TRUE(ui_test_utils::NavigateToURL(browser(), url));
 
-  auto result = content::EvalJs(main_frame(), "self.showOpenFilePicker()");
-  EXPECT_TRUE(result.error.find("self.showOpenFilePicker is not a function") !=
-              std::string::npos)
-      << result.error;
+  if (IsFileSystemAccessAPIEnabled()) {
+    auto result =
+        content::EvalJs(main_frame(), "typeof self.showOpenFilePicker");
+    EXPECT_TRUE(*result.value.GetIfString() == "function")
```
Comment (goodov) [inline renderer/test/file_system_access_browsertest.cc:78]: the dereference will crash if `GetIfString` returns `nullptr`, so we won't see `DebugString()` value at all. I'd go with:
```
ASSERT_TRUE(result.value.is_string());
EXPECT_EQ(result.value.GetString(), "function");
```
or just:
```
EXPECT_EQ(result.value.GetString(), "function");
```

**Review by goodov — APPROVED** [review]
++ with few nits

File: renderer/test/file_system_access_browsertest.cc [hunk 3]
```diff
@@ -49,14 +62,30 @@ class FileSystemAccessBrowserTest : public InProcessBrowserTest {
...
+        << "Value: " << result.value.DebugString();
+  } else {
+    auto result = content::EvalJs(main_frame(), "self.showOpenFilePicker()");
+    EXPECT_TRUE(
+        result.error.find("self.showOpenFilePicker is not a function") !=
```
Comment (iefremov) [inline renderer/test/file_system_access_browsertest.cc:83]: `base::Contains`

**Review by iefremov — APPROVED** [review]

File: chromium_src/third_party/blink/public/common/features.h (hunk 1 above)
Comment (mkarolin) [reply chromium_src/third_party/blink/public/common/features.h]: Fixed in 8aa596d

File: renderer/test/file_system_access_browsertest.cc (hunk 2 above)
Comment (mkarolin) [reply renderer/test/file_system_access_browsertest.cc]: Great catch. Fixed in 8aa596d with the second option.

File: renderer/test/file_system_access_browsertest.cc (hunk 3 above)
Comment (mkarolin) [reply renderer/test/file_system_access_browsertest.cc]: Fixed in 8aa596d

**Pull request merged**